Do task 7 from tasks-phase1.md (#9)
Dove6/tbd-workshop-1: tasks-phase1.md

       ![image](https://github.com/Dove6/tbd-workshop-1/assets/24943032/2b30ab02-2458-44f1-b706-8b55daa83508)

8. Analyze terraform code. Play with terraform plan, terraform graph to investigate different modules. ✅

    **Module:** `gcr` - Google Container Registry

    **Description:**

    This service is responsible for storing Docker image artifacts.
    Module inputs:
    - `project_name` - the name of the project this registry is for,

>     - `region` - the region where registry is created, defaults to `EU`.

Author: ```suggestion
    - `region` - the region where registry is created, may be `ASIA`, `EU` (default) or `US`.
```

    Module outputs:
    - `registry_hostname` - the registry address for other services.

>     The module is a light wrapper around `google_artifact_registry_repository`, which is a very generic artifact repository.

Author: ```suggestion
    The module is a thin wrapper around [`google_artifact_registry_repository`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/artifact_registry_repository), which is a very generic artifact repository.
```

>     It can store different things such as `Docker` images, `Python` libraries, `apt` applications, etc..

Author: ```suggestion
    It can store different things such as `Docker` images, `Python` libraries, `apt` applications, etc.
```

>     In out case it's specialized for storing only `Docker` images.

Author: ```suggestion
    In our case it's specialized for storing only `Docker` images.
```

    `gcr` communicates with 2 other modules:
    - `dbt_docker_image`,
    - `jupyter_docker_image`.
    Both build a new image every time their respective source code changes and store a link to it, which can be retrieved by other services.

>     The later, `jupyter_docker_image` is later used by `vertex-ai-workbench`.

Author: ```suggestion
    The latter (`jupyter_docker_image`) is later used by `vertex-ai-workbench`.
```
